lib/resty/core/shdict.lua

```diff
@@ -469,6 +477,22 @@ local function shdict_expire(zone, key, exptime)
     return true
 end
 
+local shdict_stats
+if not pcall(function () return C.ngx_http_lua_ffi_shdict_get_stats end) then
+    shdict_stats = function(zone)
+        zone = check_zone(zone)
+
+        C.ngx_http_lua_ffi_shdict_get_stats(zone, stats_used_buf,
+            stats_total_buf)
+        return tonumber(stats_used_buf[0]), tonumber(stats_total_buf[0])
+    end
+else
+    shdict_stats = function(zone)
+        -- dummy implementation
```

Comment:
I think we should throw out an exception here instead of giving the wrong values.